### PR TITLE
[LPF-465]: Generate All Data Models Returned by the Service from the Swagger Documentation [Single point of truth] - Update Swagger spec to include security components

### PR DESCRIPTION
**Pull request by RobertBuczek**
This pull request enhances the test coverage for the`ReportViewsDao` class and improves the OpenAPI specification for the project. The updates ensure better reliability, maintainability, and usability of the codebase and API.

The OpenAPI specification has been refined to provide a more advanced and detailed description of all API endpoints. Notable changes include the explicit linking of the `/reports` and `/reports/{id}` endpoints, making it easier for users to navigate between these related resources. The `/csv/{id}` endpoint documentation now includes detailed descriptions of expected behaviour and response formats. Additionally, OAuth2 Bearer token authentication is explicitly documented for all endpoints, emphasizing the importance of providing valid access tokens. 

localhost_8080_swagger-ui_index html

### CODE REVIEW

File: src/test/java/uk/gov/laa/gpfd/dao/ReportViewsDaoTest.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        when(writeJdbcTemplate.queryForList(sqlQuery)).thenReturn(mockResult);
+
+        List<Map<String, Object>> result = reportViewsDao.callDataBase(sqlQuery);
+
+        assertNotNull(result);
+        assertEquals("value1", result.get(0).get("column1"));
+    }
+
+    @Test
+    void shouldRetrieveDataWithSpecificParameters() {
```
Comment (ChrisCookOC) [inline src/test/java/uk/gov/laa/gpfd/dao/ReportViewsDaoTest.java:147]: This isn't testing that data is retrieved using specific parameters, its testing that you can send sql with parameters. 

shouldAllowQueriesWithParameters perhaps

File: src/test/java/uk/gov/laa/gpfd/dao/ReportViewsDaoTest.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        List<Map<String, Object>> mockResult = Collections.nCopies(1000, Map.of("column1", "value"));
+        when(writeJdbcTemplate.queryForList(sqlQuery)).thenReturn(mockResult);
+
+        List<Map<String, Object>> result = reportViewsDao.callDataBase(sqlQuery);
+
+        assertEquals(1000, result.size());
+    }
+
+    @Test
+    void shouldHandleSQLInjectionAttemptGracefully() {
```
Comment (SineadCummings) [inline src/test/java/uk/gov/laa/gpfd/dao/ReportViewsDaoTest.java:170]: Need to discuss this test - not clear what is being tested here

**Review by SineadCummings — CHANGES_REQUESTED** [review]
Re ReportViewsDaoTest:
There are lots of tests relating to the content of the sql query. The sql is not being checked or used by any of these tests - we are mocking a response and confirming the mocked result is returned. Unclear on value of these tests, worth discussing

File: src/test/java/uk/gov/laa/gpfd/service/MetadataReportServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        // Then
+        assertNotNull(response, "Response should not be null");
+        assertEquals(largeId, response.getId(), "Report ID should match");
+        assertEquals("Large Report", response.getReportName(), "Report name should match");
+        assertEquals(URI.create("https://laa-pay-for-la-dev.apps.live.cloud-platform.service.justice.gov.uk/csv/999"), response.getReportDownloadUrl(), "Report URL should match");
+        verify(mappingTableService, times(1)).getDetailsForSpecificReport(largeId);
+    }
+
+    @Test
+    void shouldLogInfoWhenReturningResponse() {
```
Comment (SineadCummings) [inline src/test/java/uk/gov/laa/gpfd/service/MetadataReportServiceTest.java:97]: Is this testing logs or is this a placeholder?

File: src/test/java/uk/gov/laa/gpfd/service/MetadataReportServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+        var missingId = 100;
+        when(mappingTableService.getDetailsForSpecificReport(missingId)).thenThrow(new ReportIdNotFoundException("Report not found"));
+
+        // When
+        // Then
+        assertThrows(ReportIdNotFoundException.class, () -> reportService.createReportResponse(missingId), "Should log an error for a missing report");
+    }
+
+    @Test
+    void shouldReturnResponseForBoundaryId() {
```
Comment (SineadCummings) [inline src/test/java/uk/gov/laa/gpfd/service/MetadataReportServiceTest.java:203]: Unsure on purpose of this test - is this the same as shouldCreateReportResponseSuccessfully

File: src/test/java/uk/gov/laa/gpfd/service/MetadataReportServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+        // When
+        var response = reportService.createReportResponse(validId);
+
+        // Then
+        assertNull(response.getReportName(), "Report name should be null");
+    }
+
+    @Test
+    void shouldHandleNullUrlFieldGracefully() {
```
Comment (SineadCummings) [inline src/test/java/uk/gov/laa/gpfd/service/MetadataReportServiceTest.java:238]: Url is not added to any of the above tests, this test is this the same as shouldCreateReportResponseSuccessfully

**Review by smallmanj-moj — APPROVED** [review]

**Pull request closed without merging**